### DISCUSSION TITLE
Do not require full environment name in tox.ini sections

### DISCUSSION THREAD

**Discussion by oakkitten** [tox-dev/tox#2434] [Ideas]
It seems that I have to have a full env name in tox.ini sections. That is, If I have an environment `my-cool-env` and I want to change an environmental variable for it, I could do:

```ini
[testenv]
setenv =
    my-cool-env: FOO = bar
```
Or I could do:

```ini
[testenv]
setenv =
    cool: FOO = bar
```

I could also do:

```ini
[testenv:my-cool-env]
setenv =
    FOO = bar
```

But this won't work:

```ini
[testenv:cool]
setenv =
    FOO = bar
```

Why not? I can't think of any downside in supporting this syntax. It seems it can simplify things a bit.

P.S. The documentation [*kind of* suggests](https://tox.wiki/en/latest/example/basic.html#ignoring-a-command-exit-code) that this syntax is possible?

**Comment by gaborbernat** [comment]
> Why not? I can't think of any downside in supporting this syntax.

This would induce ambiguity. How would you know if cool here is an env on its own or is just part of one?  The only place where we support partial names is environment markers in configurations.  I'm not sure why you conclude from the linked section that this syntax is possible 🤔

**Reply by oakkitten** [reply]
It does introduce ambiguity if you have environments `my-cool-env` and `cool`, but if you do something like that you will have ambiguity in markers as well. You can always rename `cool` to be unambiguous, which sounds like a good idea anyway. I think the benefits outweigh the cost.

Among other things, this would allow a nice workaround for #2428, instead of the following, which seems to be the only option unless you want to list all envs,

```ini
[testenv:.tox]
install_command = python -m pip install {opts} {packages}

[testenv]
install_command = your_custom_command
```
I could just write

```ini
[testenv:!.tox]
install_command = your_custom_command
```

> I'm not sure why you conclude from the linked section that this syntax is possible 🤔

Was wrong there!

**Reply by oakkitten** [reply]
I think I completely misunderstood how section names work. Apparently `[testenv:foo-{bar,baz}]` creates a single new environment `foo-{bar,baz}`. I thought that this would update environments matching `foo-bar` and `foo-baz`. Sorry!
